crypto/params.c

```diff
@@ -408,7 +418,14 @@ int OSSL_PARAM_get_int64(const OSSL_PARAM *p, int64_t *val)
         switch (p->data_size) {
         case sizeof(double):
             d = *(const double *)p->data;
-            if (d >= INT64_MIN && d <= INT64_MAX && d == (int64_t)d) {
+            if (d >= INT64_MIN
+                    /*
+		     * By subtracting 65535 (2^16-1) we cancel the low order
+		     * 15 bits of ULONG_MAX to avoid using imprecise floating
+		     * point values.
+                     */
+                    && d < (double)(INT64_MAX - 65535) + 65536.
+                    && d == (int64_t)d) {
```

Comment:
Does this also need to use real_shift()? For numbers larger than 2^real_shift(), we don't know if there was a rounding error.

Reply:
No, this is converting real to an integer, there will never be a rounding error.
`real_shift()` is for conversion from integer to real to avoid loss of lower order bits.